src/services/export.service.ts

```diff
@@ -121,4 +136,165 @@ export class ExportService {
       });
     });
   }
+
+  public generatePopulationWorksheet(
+    worksheet: ExcelJS.Worksheet,
+    testCaseGroupDto: TestCaseExcelExportDto,
+  ) {
+    let firstRow = [];
+    let headerRow = [];
+    const testCasesData = [];
+    testCaseGroupDto.testCaseExecutionResults.forEach(
+      (result: TestCaseExecutionResultDto) => {
+        const firstRowData = [];
+        const headerRowData = [];
+        const testCaseData = [];
+        const populations: PopulationDto[] =
+          this.getPopulations(testCaseGroupDto);
+        populations?.forEach((population) => {
+          firstRowData.push('Expected', 'Actual');
+          headerRowData.push(population.name, population.name);
+          this.populateTestCaseExpectedAndActual(
+            testCaseData,
+            result,
+            population,
+          );
+        });
+
+        this.populateTestCase(testCaseData, result);
+        testCasesData.push(testCaseData);
+        firstRow = firstRowData;
+        headerRow = headerRowData;
+      },
+    );
+    this.populateFirstRow(worksheet, firstRow);
+
+    this.populateHeaderRow(
+      worksheet,
+      headerRow,
+      testCaseGroupDto.testCaseExecutionResults[0],
+    );
+
+    testCasesData.forEach((testCaseData) => {
+      worksheet.addRow(testCaseData);
+    });
+    this.adjustColumnWidth(worksheet);
+  }
+
+  private getPopulations = (testCaseExcelExportDto: TestCaseExcelExportDto) => {
+    let populations: PopulationDto[] = [];
+    testCaseExcelExportDto.testCaseExecutionResults?.forEach(
+      (result: TestCaseExecutionResultDto) => {
+        if (result.populations?.length > 0) {
+          populations = result.populations;
+        }
+      },
+    );
+    return populations;
+  };
+
+  private populateTestCaseExpectedAndActual(
+    testCaseData,
+    result: TestCaseExecutionResultDto,
+    population: PopulationDto,
+  ) {
+    let foundPopulation: PopulationDto = null;
+    result.populations?.forEach((currentPopulation) => {
+      if (currentPopulation.name === population.name) {
+        foundPopulation = currentPopulation;
+      }
+    });
+    testCaseData.push(foundPopulation?.expected, foundPopulation?.actual);
+    return foundPopulation;
+  }
```

Comment:
I might be getting confused on this, but do we need to loop here if we are already looping on the populations where this is being called?

Reply:
@chubert-sb Yeah this is needed, because before this method is called, there is a filtering getPopulations() method which gets the populations that have population values, so we need to populate the real values from the original dto.